go/nomdl/lexer.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package nomdl
+
+import (
+	"fmt"
+	"text/scanner"
+)
+
+type token struct {
+	Pos  scanner.Position
+	Type rune
+	Text string
+}
+
+func (t token) String() string {
+	return t.Text
+}
+
+type lexer struct {
+	scanner   *scanner.Scanner
+	nextToken token
```

Comment:
On my first readthrough, it seems like `nextToken` is often redundant with state that's stored inside `scanner`. Can you provide some comments explaining what you needed it for? Up here is fine, or wherever it is in this file that you take advantage of copying the state out of scanner and storing it here.